neutron/neutron.go

```diff
@@ -51,12 +58,45 @@ func New(client client.Client) *Client {
 	return &Client{client}
 }
 
-// ListNetworksV2 gives details on available networks
-func (c *Client) ListNetworksV2() ([]NetworkV2, error) {
+// ----------------------------------------------------------------------------
+// TODO(hml): copied from the nova package.  However it should really be pulled out
+// and shared between goose pkgs, however we don't want to break compatibility at this time.
+//
+// Filtering helper.
+//
+// Filter builds filtering parameters to be used in an OpenStack query which supports
+// filtering.  For example:
+//
+//     filter := NewFilter()
+//     filter.Set(neutron.FilterRouterExternal, "true")
+//     resp, err := neutron.ListNetworks(filter)
+//
+type Filter struct {
+	v url.Values
+}
+
+// NewFilter creates a new Filter.
+func NewFilter() *Filter {
+	return &Filter{make(url.Values)}
+}
+
+func (f *Filter) Set(filter, value string) {
+	f.v.Set(filter, value)
+}
+
+// ----------------------------------------------------------------------------
+
+// ListNetworksV2 gives details on available networks, zero or one Filters
```

Comment:
Yes, at the moment, we're trying avoid bumping the goose version to get Bug 1654144 fixed in juju 2.2.  The plan is to bump to to v2 in the next month or so, to go with other changes needed for the next release of juju.